### ISSUE TITLE
Editor handling

### ISSUE THREAD

**Issue by lokothodida** [lokothodida/gs-catalog#1]
_CKEditor_ and _CodeMirror_ are likely to undergo modularisation in the GS Core in the future. Currently both editors are handled with code copy-pasted from the core, but it may be best to do so separately instead.

It is relatively simply to have the catalog editor logic completely separate from the rest of the GS installation and only provide what is needed for the catalog itself. Perhaps have editor options in the **Options** panel?

**Comment by Tzvook** [comment]
Yes, editor options in the Options panel is the best option, I fixed it since my GS site has it's own editor options and it broke the wysiwyg.

**Comment by lokothodida** [comment]
What sorts of options should be available?
- GS's advanced options?
- ...

Also, it _might_ be possible to add a small CKEditor plugin that allows the admin to more easily embed a link to an existing product/category (but I don't want to get too ahead of myself here).

**Comment by lokothodida** [comment]
Current v0.2 branch has the following:

## Editor enabled

![selection_001](https://cloud.githubusercontent.com/assets/4363863/4057830/f06267d2-2dcd-11e4-9567-7d0ea69b6721.png)

## Editor disabled

![selection_002](https://cloud.githubusercontent.com/assets/4363863/4057841/0c3ab266-2dce-11e4-94bd-85bcce72c7ee.png)

Will need to add in a few more i18n hashes for the language file for this change (and likely others).

**Comment by Tzvook** [comment]
Looks good, I'll test it tomorrow ..
You saw what I wrote about the image-picker (it's in my fork, at the 0.1 - https://github.com/Tzvook/gs-catalog ) ? I added "Image" field to the custom-fields , but it probably need to be done to the category images too ..

**Comment by lokothodida** [comment]
~~I saw it, and will integrate the changes soon as well. Are you making any more changes to the plugin at the moment? Because it might be best to merge your changes in soon/now before my changes diverge too hugely from the current master. It's possible also that v0.2 is released as a pre-production beta before further changes are made, and that the changes are easier to document.~~

I've merged your changes into the master branch on the main repository, and I am going to merge v0.2 in soon. When that is done, update your master branch (as stated by the asterisk [here](https://github.com/lokothodida/gs-catalog/wiki/Contributing%20to%20the%20codebase)), create a new branch for the changes you want to make, and work on that branch until you've done your changes and I'll merge them in (and the cycle continues any other time you want to make changes to the catalog plugin).
